<robot name="trrp_robot">
  <link name="base_link">
    <visual>
      <geometry>
        <mesh filename="package://urdf_robots/mesh/base_link_2p.STL"/>
      </geometry>
      <origin rpy="0 0 0" xyz="-0.1 -0.1 0"/>
      <material name="generic">
        <color rgba="1 1 0 1"/>
      </material>
    </visual>
  </link>

  <joint name="joint1" type="revolute">
    <parent link="base_link"/>
    <child link="link1"/>
    <origin xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="300" velocity="0.1" lower="-3.14" upper="3.14"/>
    <dynamics damping="50" friction="1"/>
  </joint>

  <link name="link1">
    <visual>
      <geometry>
        <mesh filename="package://urdf_robots/mesh/link1_2p.STL"/>
      </geometry>
      <origin rpy="0 0 1.274090353" xyz="0.115 -0.08 0"/>
      <material name="generic_2">
        <color rgba="0 0 1 1"/>
      </material>
    </visual>
  </link>

  <joint name="joint2" type="revolute">
    <parent link="link1"/>
    <child link="link2"/>
    <origin xyz="0.1 -0.05 0.3"/>
    <axis xyz="0 -1 0"/>
    <limit effort="300" velocity="0.1" lower="-0.75" upper="0.75"/>
    <dynamics damping="50" friction="1"/>
  </joint>

  <link name="link2">
    <visual>
      <geometry>
        <mesh filename="package://urdf_robots/mesh/link2_2p.STL"/>
      </geometry>
      <origin rpy="0 0 3.14159265" xyz="0.215 0.025 -0.03"/>
      <material name="generic_3">
        <color rgba="0 1 0 1"/>
      </material>
    </visual>
  </link>

  <joint name="joint3" type="revolute">
    <parent link="link2"/>
    <child link="link3"/>
    <origin xyz="0.2 -0.05 0"/>
    <axis xyz="0 -1 0"/>
    <limit effort="300" velocity="0.1" lower="-0.15" upper="0.5"/>
    <dynamics damping="50" friction="1"/>
  </joint>
  <link name="link3">
    <visual>
      <geometry>
        <mesh filename="package://urdf_robots/mesh/link3_2p.STL"/>
      </geometry>
      <origin rpy="0 0 0" xyz="-0.0225 -0.025 -0.0225"/>
      <material name="generic_4">
        <color rgba="0 1 1 1"/>
      </material>
    </visual>
  </link>

  <joint name="joint4" type="prismatic">
    <parent link="link3"/>
    <child link="link4"/>
    <origin xyz="0.15 0 0"/>
    <axis xyz="0 0 -1"/>
    <limit effort="300" velocity="0.1" lower="0" upper="0.2"/>
    <dynamics damping="50" friction="1"/>
  </joint>
  <link name="link4">
    <visual>
      <geometry>
        <mesh filename="package://urdf_robots/mesh/link4_2p.STL"/>
      </geometry>
      <origin rpy="0 3.14159265 0" xyz="0.015 -0.015 0.2"/>
      <material name="generic_5">
        <color rgba="1 0 0 1"/>
      </material>
    </visual>
  </link>

</robot>
